include/asterisk/channel.h

```diff
@@ -3119,48 +3121,65 @@ struct ast_channel *ast_channel_iterator_next(struct ast_channel_iterator *i);
 
 /*! @} End channel iterator definitions. */
 
+/*! @{ Channel search functions */
+
+/*!
+* \warning If the legacy ao2_container channel storage backend is in use,
```

Comment:
This should just state that no channel locks be held. Since this is a developer comment, and the selection of the backend is a user runtime selection you can't really reconcile the two and take it into account.

Reply:
Done